File: Leetcode/code/3.py
def lengthOfLongestSubstring( s: str) -> int:
    i = 0 
    longest = 0

    d = {}
    
    while i < len(s):
        count = 0 
        for j in range(i, len(s)):
            if s[j] not in d:
                d[s[j]] = 1
                count += 1
            else: 
                break
        d = {}
        if count > longest:
            longest = count
        
        
        i += 1

    print(longest)
    return longest
        

# 문자가 다른 가장 긴 부분문자열을 찾으래 ㅠㅡㅠ


lengthOfLongestSubstring("abcabcbb")

File: leetcode/3.js
function lengthOfLongestSubstring(s) {
    const map = {};
    var left = 0;
    
    return s.split('').reduce((max, v, i) => {
        console.log(max, v, i)
        console.log(map[v])
        left = map[v] >= left ? map[v] + 1 : left;
        console.log(left, "left값")
        map[v] = i;
        
        return Math.max(max, i - left + 1);
    }, 0);
}


lengthOfLongestSubstring('abcabcbb')